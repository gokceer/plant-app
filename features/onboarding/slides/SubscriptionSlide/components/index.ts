export { default as FeatureHeader } from "./FeatureHeader/FeatureHeader";
export { default as FeatureList } from "./FeatureList/FeatureList";
export { default as FooterLinks } from "./FooterLinks/FooterLinks";
export { default as PlanOptions } from "./PlanOptions/PlanOptions";
export { default as SubscriptionNote } from "./SubscriptionNote/SubscriptionNote";
