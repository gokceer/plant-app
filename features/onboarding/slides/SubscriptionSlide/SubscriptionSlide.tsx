import { MainButton } from "@/components";
import React from "react";
import {
  Dimensions,
  Image,
  Pressable,
  ScrollView,
  StyleSheet,
  View,
} from "react-native";
import {
  FeatureHeader,
  FeatureList,
  FooterLinks,
  PlanOptions,
  SubscriptionNote,
} from "./components";

const { width: screenWidth } = Dimensions.get("window");
const aspectRatio = 1125 / 1470;
const imageHeight = screenWidth / aspectRatio;

const SubscriptionSlide = ({
  onCloseBtnPress,
  onTryFreeBtnPress,
}: {
  onCloseBtnPress: () => void;
  onTryFreeBtnPress?: () => void;
}) => {
  return (
    <ScrollView
      contentContainerStyle={styles.scrollContainer}
      style={styles.container}
      showsVerticalScrollIndicator={false}
      bounces={false}
    >
      <Image
        source={require("../../../../assets/images/paywallPlant.png")}
        style={styles.topImage}
        resizeMode="cover"
      />
      <Pressable
        style={styles.closeButton}
        onPress={onCloseBtnPress}
        testID="closeButton"
      >
        <Image
          source={require("../../../../assets/icons/closeIcon.png")}
          style={{ width: 24, height: 24 }}
        />
      </Pressable>
      <View style={styles.content}>
        <FeatureHeader />
        <FeatureList />
        <PlanOptions />
        <MainButton
          title="Try free for 3 days"
          onPress={() => onTryFreeBtnPress?.()}
        />
        <SubscriptionNote />
        <FooterLinks />
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  scrollContainer: {
    paddingBottom: 20,
  },
  container: {
    flex: 1,
    position: "relative",
    backgroundColor: "#101e17",
    width: screenWidth,
  },
  topImage: {
    width: screenWidth,
    height: imageHeight,
    position: "absolute",
    top: 0,
    left: 0,
    right: 0,
  },
  content: {
    flex: 1,
    paddingTop: "70%",
    paddingHorizontal: 24,
  },
  closeButton: {
    position: "absolute",
    top: 55,
    right: 20,
  },
});

export default SubscriptionSlide;
