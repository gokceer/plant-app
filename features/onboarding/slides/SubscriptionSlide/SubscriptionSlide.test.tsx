import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import SubscriptionSlide from "./SubscriptionSlide";

describe("SubscriptionSlide", () => {
  it("renders correctly and matches snapshot", () => {
    const mockOnClose = jest.fn();

    const { toJSON } = render(
      <SubscriptionSlide onCloseBtnPress={mockOnClose} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it("calls onCloseBtnPress when close is pressed", () => {
    const mockOnClose = jest.fn();

    const { getByTestId } = render(
      <SubscriptionSlide onCloseBtnPress={mockOnClose} />
    );

    const closeBtn = getByTestId("closeButton");
    fireEvent.press(closeBtn);

    expect(mockOnClose).toHaveBeenCalledTimes(1);
  });

  it("calls onTryFreeBtnPress when pressed", () => {
    const mockOnClose = jest.fn();
    const mockOnTryFree = jest.fn();
    const { getByTestId, getByText } = render(
      <SubscriptionSlide
        onCloseBtnPress={mockOnClose}
        onTryFreeBtnPress={mockOnTryFree}
      />
    );

    expect(getByText("Try free for 3 days")).toBeTruthy();

    const tryFreeBtn = getByTestId("mainButton");
    fireEvent.press(tryFreeBtn);

    expect(mockOnTryFree).toHaveBeenCalledTimes(1);
  });
});
