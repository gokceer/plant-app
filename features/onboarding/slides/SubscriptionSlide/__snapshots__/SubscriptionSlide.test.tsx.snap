// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscriptionSlide renders correctly and matches snapshot 1`] = `
<RCTScrollView
  bounces={false}
  contentContainerStyle={
    {
      "paddingBottom": 20,
    }
  }
  showsVerticalScrollIndicator={false}
  style={
    {
      "backgroundColor": "#101e17",
      "flex": 1,
      "position": "relative",
      "width": 750,
    }
  }
>
  <View>
    <Image
      resizeMode="cover"
      source={
        {
          "testUri": "../../../assets/images/paywallPlant.png",
        }
      }
      style={
        {
          "height": 980,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "position": "absolute",
          "right": 20,
          "top": 55,
        }
      }
      testID="closeButton"
    >
      <Image
        source={
          {
            "testUri": "../../../assets/icons/closeIcon.png",
          }
        }
        style={
          {
            "height": 24,
            "width": 24,
          }
        }
      />
    </View>
    <View
      style={
        {
          "flex": 1,
          "paddingHorizontal": 24,
          "paddingTop": "70%",
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "Rubik",
            "fontSize": 26,
            "marginBottom": 8,
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "RubikBold",
              "fontSize": 30,
            }
          }
        >
          PlantApp
        </Text>
         Premium
      </Text>
      <Text
        style={
          {
            "color": "#FFFFFFB2",
            "fontFamily": "RubikLight",
            "fontSize": 17,
          }
        }
      >
        Access All Features
      </Text>
      <View
        style={
          {
            "height": 150,
            "paddingTop": 20,
          }
        }
      >
        <RCTScrollView
          bounces={false}
          horizontal={true}
          showsHorizontalScrollIndicator={false}
        >
          <View>
            <View
              style={
                {
                  "backgroundColor": "#FFFFFF14",
                  "borderRadius": 12,
                  "height": 130,
                  "marginRight": 8,
                  "padding": 16,
                  "width": 156,
                }
              }
            >
              <Image
                source={
                  {
                    "testUri": "../../../assets/icons/scanIcon.png",
                  }
                }
                style={
                  {
                    "height": 36,
                    "width": 36,
                  }
                }
              />
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontFamily": "RubikMedium",
                    "fontSize": 20,
                    "paddingTop": 16,
                  }
                }
              >
                Unlimited
              </Text>
              <Text
                style={
                  {
                    "color": "#FFFFFFB2",
                    "fontFamily": "RubikRegular",
                    "fontSize": 12,
                    "paddingTop": 4,
                  }
                }
              >
                Plant Identify
              </Text>
            </View>
            <View
              style={
                {
                  "backgroundColor": "#FFFFFF14",
                  "borderRadius": 12,
                  "height": 130,
                  "marginRight": 8,
                  "padding": 16,
                  "width": 156,
                }
              }
            >
              <Image
                source={
                  {
                    "testUri": "../../../assets/icons/fastIcon.png",
                  }
                }
                style={
                  {
                    "height": 36,
                    "width": 36,
                  }
                }
              />
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontFamily": "RubikMedium",
                    "fontSize": 20,
                    "paddingTop": 16,
                  }
                }
              >
                Faster
              </Text>
              <Text
                style={
                  {
                    "color": "#FFFFFFB2",
                    "fontFamily": "RubikRegular",
                    "fontSize": 12,
                    "paddingTop": 4,
                  }
                }
              >
                Process
              </Text>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        style={
          {
            "marginTop": 20,
          }
        }
      >
        <View
          accessibilityRole="radio"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": false,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "borderRadius": 14,
                "marginBottom": 16,
                "padding": 16,
              },
              {
                "backgroundColor": "#FFFFFF0D",
              },
            ]
          }
          testID="planCardmonthly"
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "gap": 12,
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF26",
                    "borderRadius": 12,
                    "height": 24,
                    "justifyContent": "center",
                    "width": 24,
                  },
                  false,
                ]
              }
              testID="radioOuterView"
            />
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "position": "relative",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#fff",
                      "fontFamily": "RubikMedium",
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  1 Month
                </Text>
              </View>
              <Text
                style={
                  {
                    "color": "#FFFFFFB2",
                    "fontFamily": "RubikLight",
                    "fontSize": 14,
                    "marginTop": 4,
                  }
                }
              >
                $2.99/month, auto renewable
              </Text>
            </View>
          </View>
        </View>
        <View
          accessibilityRole="radio"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": true,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "borderRadius": 14,
                "marginBottom": 16,
                "padding": 16,
              },
              {
                "borderColor": "#28AF6E",
                "borderWidth": 1.5,
              },
            ]
          }
          testID="planCardyearly"
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "gap": 12,
              }
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF26",
                    "borderRadius": 12,
                    "height": 24,
                    "justifyContent": "center",
                    "width": 24,
                  },
                  {
                    "backgroundColor": "#28AF6E",
                  },
                ]
              }
              testID="radioOuterView"
            >
              <View
                style={
                  {
                    "backgroundColor": "white",
                    "borderRadius": 4,
                    "height": 8,
                    "width": 8,
                  }
                }
                testID="radioInnerView"
              />
            </View>
            <View
              style={
                {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "position": "relative",
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#fff",
                      "fontFamily": "RubikMedium",
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  1 Year
                </Text>
                <View
                  style={
                    {
                      "backgroundColor": "#34C759",
                      "borderBottomLeftRadius": 20,
                      "borderTopRightRadius": 12,
                      "paddingHorizontal": 12,
                      "paddingVertical": 4,
                      "position": "absolute",
                      "right": -16,
                      "top": -16,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#fff",
                        "fontFamily": "RubikMedium",
                        "fontSize": 12,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Save 50%
                  </Text>
                </View>
              </View>
              <Text
                style={
                  {
                    "color": "#FFFFFFB2",
                    "fontFamily": "RubikLight",
                    "fontSize": 14,
                    "marginTop": 4,
                  }
                }
              >
                First 3 days free, then $529.99/year
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#28AF6E",
            "borderRadius": 12,
            "paddingVertical": 16,
          }
        }
        testID="mainButton"
      >
        <Text
          style={
            {
              "color": "#fff",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
        >
          Try free for 3 days
        </Text>
      </View>
      <Text
        style={
          {
            "color": "#FFFFFF85",
            "fontFamily": "RubikLight",
            "fontSize": 10,
            "marginTop": 8,
            "textAlign": "center",
          }
        }
      >
        After the 3-day free trial period you’ll be charged ₺274.99 per year unless you cancel before the trial expires. Yearly Subscription is Auto-Renewable
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 14,
              "marginHorizontal": 6,
            }
          }
        >
          Terms
        </Text>
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 14,
            }
          }
        >
          •
        </Text>
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 14,
              "marginHorizontal": 6,
            }
          }
        >
          Privacy
        </Text>
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 14,
            }
          }
        >
          •
        </Text>
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 14,
              "marginHorizontal": 6,
            }
          }
        >
          Restore
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
